data/projects.js

```diff
@@ -147,5 +147,133 @@ coala should support generating metrics for your code.",
 		"initiatives" : ["GSoC"],
 		"tags" : ["Web", "Blog"],
 		"markdown": "coala_web.md"
+	},
+	{
+		"name" : "Optimize Caching",
+		"desc" : "Improve coala's performance by implementing caching strategies.",
+		"requirements" : [
+			"At least one patch to the coala core should be accepted and merged.",
+			"The student should be familiar with ongoing technical cEPs.",
+			"The proposal needs to show some familiarity with coala's caching \
+			implementation.",
+			"The student should have understanding of the current caching mechanism"
+		],
+		"difficulty" : "medium",
+		"issues" : [
+			"https://github.com/coala/coala/issues/2541",
+			"https://github.com/coala/coala/issues/2699",
+			"https://github.com/coala/coala/issues/2912"
+		],
+		"mentors" : [
+			"adtac",
+			"sils"
+		],
+		"initiatives" : ["GSoC"],
+		"tags" : ["Core", "Bears"],
+		"markdown": "optimize_caching.md"
+	},
+	{
+		"name" : "Cohesive Editor Support",
+		"desc" : "",
+		"requirements" : [
+			"The proposal has to provide a convincing strategy for testing \
+			editor plugins",
+			"The applicant should have one patch accepted to any of the coala \
+			projects."
+		],
+		"difficulty" : "medium",
+		"issues" : [
+			"https://github.com/coala/coala-atom/issues/37",
+			"https://github.com/coala/coala-emacs/issues/11",
+			"https://github.com/coala/coala-eclipse/issues/25",
+			"https://github.com/coala/coala-vim/issues/24",
+			"https://github.com/coala/coala-gedit/issues/5",
+			"https://github.com/coala/coala-sublime/issues/4"
+		],
+		"mentors" : [
+			"AbdealiJK",
+			"fneu"
+		],
+		"initiatives" : ["GSoC"],
+		"tags" : ["Editors", "Plugins"],
+		"markdown": "editor_support.md"
+	},
+	{
+		"name" : "Python dead code fixer",
+		"desc" : "Wrap two flake8 AST plugins and write a native bear for dead \
+		code detection and fixing for python.",
+		"requirements" : [
+			"At least one patch to the coala core should be accepted and merged.",
+		],
+		"difficulty" : "medium",
+		"issues" : [],
+		"mentors" : [
+			"jayvdb"
+		],
+		"initiatives" : ["GSoC"],
+		"tags" : ["Bears"],
+		"markdown": "dead_python.md.md"
+	},
+	{
+		"name" : "Real-time team communication bridge",
+		"desc" : "A way to post CI problems to online team rooms, such as IRC, \
+		Telegram, slack or gitter.",
+		"requirements" : [
+			"At least one patch to the coala core should be accepted and merged.",
+		],
+		"difficulty" : "medium",
+		"issues" : [],
+		"mentors" : [
+			"jayvdb"
+		],
+		"initiatives" : ["GSoC"],
+		"tags" : ["CI"],
+		"markdown": "communication_bridge.md"
+	},
+	{
+		"name" : "Integrating Antlr v4 into coala",
+		"desc" : "Integrate Antlr v4 into coala to provide bears an AST \
+		interface",
+		"requirements" : [
+			"At least one patch to the coala core should be accepted and merged.",
+		],
+		"difficulty" : "hard",
+		"issues" : [],
+		"mentors" : [
+			"jayvdb"
```

Comment:
Would be good for others to take this over.
Especially good for Java programmers in our midst.